File: com/teamiurg/techexpansion/block/TEBlock.java
package com.teamiurg.techexpansion.block;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.creativetab.CreativeTabs;

public class TEBlock extends Block {

	public TEBlock(Material mat, String name, CreativeTabs tab, float hardness, float resistance, SoundType stepSound,
			int lightOpacity, float lightLevel, String toolTypeRequired, int toolLevelRequired, boolean tickRandomly) {
		super(mat);
		this.setUnlocalizedName(name).setHardness(hardness).setResistance(resistance).setStepSound(stepSound)
				.setLightOpacity(lightOpacity).setLightLevel(lightLevel).setCreativeTab(tab)
				.setHarvestLevel(toolTypeRequired, toolLevelRequired);
		this.setTickRandomly(tickRandomly);
	}

}

File: src/main/scala/com/teamisotope/techexpansion/block/TEBlock.scala
package com.teamisotope.techexpansion.block

import net.minecraft.block._
import net.minecraft.block.material.Material
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.ItemBlock

class TEBlock(mat: Material, name: String, tab: CreativeTabs, hardness: Float, resistance: Float, toolType: String, toolLevel: Int) extends Block(mat) {
  this.setUnlocalizedName(name).setRegistryName(name).setCreativeTab(tab).setHardness(hardness).setResistance(resistance).setHarvestLevel(toolType, toolLevel);
  val itemblock: ItemBlock = new ItemBlock(this)
  itemblock.setRegistryName(name).setUnlocalizedName(name)
}
